hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -1637,18 +1637,15 @@ class Emit[C](
               val updateMap = Map(fields: _*)
               val srvb = new StagedRegionValueBuilder(mb, x.pType, region.code)
 
-              val addFields = { (newMB: EmitMethodBuilder[C], t: PType, v: EmitCode) =>
-                Code(
-                  v.setup,
-                  v.m.mux(srvb.setMissing(), srvb.addIRIntermediate(t)(v.v)),
-                  srvb.advance())
-              }
+              def addFields(t: PType, v: EmitCode): Code[Unit] = Code(
```

Comment:
why these changes? Did something break?

Reply:
The `emitSelf.emit` call is bad, because it will use the Region argument of `mb` rather than the region passed to emit. This looked to me like some cruft left from the method splitter change, so I cleaned it up while I was fixing the bug. I don't remember anything breaking, I think I found this by looking at where the `emit` overload that uses the region argument is used.

Reply:
ahhhh, got it.